unlock-app/src/__tests__/services/unlockProvider.test.js

```diff
@@ -1,40 +1,196 @@
-    expect.assertions(2)
-    provider.setAddress('0x12345678')
-    provider.send(rpc('eth_accounts'), (_, data) => {
-      expect(data.result).toHaveLength(1)
-      expect(data.result[0]).toBe('0x12345678')
-      done()
+  describe('signing data', () => {
+    describe('signUserData', () => {
+      it('should sign an object with all fields passed', () => {
+        expect.assertions(1)
+        const input = {
+          emailAddress,
+          publicKey,
+          passwordEncryptedPrivateKey: key,
+        }
+
+        const output = provider.signUserData(input)
+        // sigutil seems to downcase things
+        expect(sigUtil.recoverTypedSignature(output)).toEqual(
+          publicKey.toLowerCase()
+        )
+      })
+
+      it('should also sign an object with default values when not everything is passed', () => {
+        expect.assertions(1)
+        const input = {}
+
+        const output = provider.signUserData(input)
+        // sigutil seems to downcase things
+        expect(sigUtil.recoverTypedSignature(output)).toEqual(
+          publicKey.toLowerCase()
+        )
+      })
+    })
+
+    describe('signPaymentData', () => {
+      it('should sign a stripe card token', () => {
+        expect.assertions(1)
+        const token = 'tok_1EPsocIsiZS2oQBMRXzw21xh'
+        const output = provider.signPaymentData(token)
+
+        expect(sigUtil.recoverTypedSignature(output)).toEqual(
+          publicKey.toLowerCase()
+        )
+      })
+    })
+
+    describe('signKeyPurchaseRequestData', () => {
+      it('should sign a key purchase request with a valid expiration time', () => {
+        expect.assertions(2)
+        const input = {
+          recipient: publicKey,
+          lock: '0xaC6b4470B0cba92b823aB96762972e67a1C851d5',
+        }
+        const output = provider.signKeyPurchaseRequestData(input)
+        const currentTime = Math.floor(Date.now() / 1000)
+
+        expect(sigUtil.recoverTypedSignature(output)).toEqual(
+          publicKey.toLowerCase()
+        )
+        // Expiration must be some amount of time after right now
+        expect(
+          output.data.message.purchaseRequest.expiry > currentTime
+        ).toBeTruthy()
+      })
+    })
+
+    describe('personal_sign', () => {
+      it('should sign some hex data with the user account private key', () => {
+        expect.assertions(1)
+        const someData = 'this is the data I want to sign'
+        const messageHash = utf8ToHex(someData)
+
+        // second param is unused, but in keeping with what we receive from WalletService
+        const sig = provider.personal_sign([messageHash, ''])
+
+        expect(
+          sigUtil.recoverPersonalSignature({
+            data: messageHash,
+            sig,
+          })
+        ).toEqual(publicKey.toLowerCase())
+      })
+
+      it('should be compatible with walletService', async done => {
```

Comment:
Weird that this test failed

Reply:
Probably some network call. trying to see why.

Reply:
What's weirder is that it passes locally.

Reply:
I found it!

Reply:
What was the problem?

Reply:
We are making a network call to get the network id! 

But apparently there is smething else.